TM1py/Services/CellService.py

```diff
@@ -568,6 +568,96 @@ def clear(self, cube: str, **kwargs):
 
         return self.clear_with_mdx(cube=cube, mdx=mdx_builder.to_mdx(), **kwargs)
 
+    @require_data_admin
+    @require_ops_admin
+    @require_version(version="11.7")
+    def clear_from_df(self, cube: str, df: pd.DataFrame, dimension_mapping: Dict = None, **kwargs):
+        """Clears data from a TM1 cube based on the distinct values in a DataFrame over cube dimensions.
+            Note:
+                This function is similar to `tm1.cells.clear`, but it is designed specifically for clearing data
+                 based on distinct values in a DataFrame over cube dimensions. The key difference is that this
+                 function interprets the DataFrame columns as dimensions and supports a mapping (`dimension_mapping`)
+                 for specifying hierarchies within those dimensions.
+
+          :param cube: str
+              The name of the TM1 cube.
+          :param df: pd.DataFrame
+              The DataFrame containing distinct values over cube dimensions.
+              Columns in the DataFrame should correspond to cube dimensions.
+          :param dimension_mapping: Dict, optional
+              A dictionary mapping DataFrame columns to specific hierarchies within TM1 dimensions.
+              If not provided, assumes that the dimensions have just one hierarchy.
+
+          :return: None
+              The function clears data in the specified TM1 cube.
+
+          :raises DxdException:
+              If there are unmatched dimensions in the DataFrame or if specified dimensions
+              do not exist in the TM1 cube.
+
+          :example:
+              ```python
+
+              # Sample DataFrame with distinct values over cube dimensions
+              data = {
+                  "Year": ["2021", "2022"],
+                  "Organisation": ["some_company", "some_company"],
+                  "Location": ["Germany", "Albania"]
+              }
+
+              # Sample dimension mapping
+              dimensions_mapping = {
+                  "Organisation": "hierarchy_1",
+                  "Location": ["hierarchy_2", "hierarchy_3", "hierarchy_4"]
+              }
+              ```
+        """
+
+        if not dimension_mapping:
+            dimension_mapping = {}
+
+        cube_service = self.get_cube_service()
+        dimension_names = CaseAndSpaceInsensitiveSet(*cube_service.get_dimension_names(cube_name=cube))
+
+        df = df.astype(str)
+
+        unique_entries = [{col_name: df[col_name].unique()} for col_name in df.columns]
+
+        cell_coordinates = {}
+        unmatched_dimension_names = []
+        for entry in unique_entries:
+            coordinates_list = []
+            for key, value in entry.items():
+                if key not in dimension_names:
+                    unmatched_dimension_names.append(key)
+                for i in value:
+                    if key in dimension_mapping:
+                        coordinates_list.append(f"[{key}].[{dimension_mapping[key]}].[{i}]")
```

Comment:
If we used `mdxpy.Member.of(...).to_mdx()` here instead of building the unique element name ourselves, it would take care of the rare edge case of escaping the `]` in case an element in the data frame contains `]`

Reply:
Hi @MariusWirtz,
I wanted to try it out, but I think you mean `mdxpy.Member.of(...).unique_name`

If found this one here:
```
element_definition = Member.of(dimension_name, hierarchy_name, element_name)
mdx_strings_list.append("{" + element_definition.unique_name + "}")
```

I think, we need to use mdxpy for this part as well:https://github.com/cubewise-code/tm1py/blob/fdcb07b0fd975e9b8be43cc010c37942dc4c9123/TM1py/Services/CellService.py#L643

Reply:
Yes. You are right on both points. 
Thank you